Character.cpp

```diff
@@ -98,20 +97,37 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        if ( initialArmorLevel && initialHitPoints && initialAttackDamage ) 
+        {
+            initialHitPoints.reset( new int( *initialHitPoints ) );
+            initialArmorLevel.reset( new int( *initialArmorLevel) );
+            initialAttackDamage.reset( new int( *initialAttackDamage ) );
+            boostArmor(getArmorLevel() * 0.1);
+            boostHitPoints(getHP() * 0.1);
+            boostAttackDamage(getAttackDamage() * 0.1);
+            *initialArmorLevel = getArmorLevel();
+            *initialHitPoints = getHP();
+            *initialAttackDamage = getAttackDamage();
```

Comment:
All of this is unnecessary.  You can access `hitPoints` and `armor` and `attack` directly within the class. 
You don't need to use any of these public getters to modify those variables.